scarb/src/bin/scarb/args.rs

```diff
@@ -224,6 +224,18 @@ pub struct BuildArgs {
     /// Build tests.
     #[arg(short, long, default_value_t = false)]
     pub test: bool,
+
+    /// Comma separated list of features to activate
+    #[arg(short = 'F', long, value_delimiter = ',', env = "SCARB_FEATURES")]
+    pub features: Vec<String>,
+
+    /// Activate all available features
+    #[arg(long, default_value_t = false, env = "SCARB_ALL_FEATURES")]
+    pub all_features: bool,
+
+    /// Do not activate the `default` feature
+    #[arg(long, default_value_t = false, env = "SCARB_NO_DEFAULT_FEATURES")]
+    pub no_default_features: bool,
```

Comment:
Should we ensure `all_features` and `no_default_features` are not called simultaneously? You can do this with `group` in Clap.

Reply:
In rust, calling both doesn't result in any error, because `--all-features` takes priority and that's how I implemented it. It may seem contradictory at first, but I think it makes sense, because `--no-default-features` just tells "don't enable `default` feature artificially", while `--all-features` enables everything. The same way, you can have both feature `x` enabled and `--no-default-features`, even if `x` is a default feature.

Reply:
> The same way, you can have both feature x enabled and --no-default-features, even if x is a default feature.

Actually, I think this is useful, as you can set `x,y,z` as default, and only enable `x` with `--features`. On the other hand `--all-features` + `--no-default-features` just ignores one of the options 🤔

Personally, I would diverge from Cargo behaviour and ban using both `--all-features` and `--no-default-features` simultaneously, unless we have a use case for that.

Reply:
Hmm, I just don't see why? Also, if we can rely on Clap for args parsing / validation - we should.

Reply:
Ok, will add that validation